src/miner.cpp

```diff
@@ -106,110 +105,215 @@ void UpdateTime(CBlockHeader* pblock, const Consensus::Params& consensusParams,
     pblock->nTime = std::max(pindexPrev->GetMedianTimePast()+1, GetAdjustedTime());
 }
 
-void GetBlockTxPriorityData(const CBlock *pblock, int nHeight, int64_t nMedianTimePast, const CCoinsViewCache& view,
-                               vector<TxPriority>& vecPriority, list<COrphan>& vOrphan, map<uint256, vector<COrphan*> >& mapDependers)
+bool HandleScDependancy(const CTransactionBase& tx, const CCoinsViewCache& view, COrphan*& porphan, list<COrphan>& vOrphan, map<uint256, vector<COrphan*> >& mapDependers)
 {
-    for (map<uint256, CTxMemPoolEntry>::iterator mi = mempool.mapTx.begin();
-         mi != mempool.mapTx.end(); ++mi)
+    // detect dependancies from the sidechain point of view 
+    for (const auto& ft: tx.GetVftCcOut())
```

Comment:
Do we need to check here also sc creation outputs to not to include 2 or more Txs, that try to create SC with the same id? Or is it somehow prevented while applying to the mempool?